src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/CSharpFormatter.cs

```diff
@@ -273,6 +273,17 @@ static bool ShouldIgnoreLineCompletelyBecauseOfAncestors(SyntaxToken token, Sour
                     return false;
                 }
 
+                // When directly in an implicit object creation expression, it seems the C# formatter
+                // does format the braces of an array initializer, so we need to special case those
+                // node types. Doing it outside the loop is good for perf, but also makes things easier.
+                if (parent is InitializerExpressionSyntax initializer &&
+                    initializer.IsKind(CodeAnalysis.CSharp.SyntaxKind.ArrayInitializerExpression) &&
+                    (token == initializer.OpenBraceToken || token == initializer.CloseBraceToken) &&
+                    initializer.Parent?.Parent?.Parent?.Parent is ImplicitObjectCreationExpressionSyntax)
```

Comment:
lol, gotta love the `Parent?.Parent?.Parent?.Parent` 😄

Reply:
lol, do we have an ancestor lookup extension method or something or would that do the wrong thing?

Reply:
We wouldn't want to use the `AncestorsAndSelf()` method here because we're not looking for anything that is parented by an implicit object creation expression, but rather a specific tree structure. It does look funny though :)

Reply:
lol geeezzz